src/main/webapp/search-script.js

```diff
@@ -0,0 +1,122 @@
+function collectUnitsFromSearch() {
+  const inputElems = document.getElementsByName("search-units");
+  const addedUnits = []
+  for (let elem of inputElems) {       
+    if (elem.checked){
+      addedUnits.push(elem.value);
+    }
+  }
+  return addedUnits;
+}
+
+function collectUnitsFromInput() {
+  const inputElems = document.getElementsByName("num-units");
+  const addedUnits = []
+  for (let elem of inputElems) {       
+    if (elem.checked){
+      addedUnits.push(elem.value);
+    }
+  }
+  return addedUnits;
```

Comment:
Seems everything but the inputElems is identical for these two functions. Consider creating a utility function instead that takes in the inputElems as a parameter